definition_validator/validator.cc

```diff
@@ -901,6 +905,28 @@ class ValidateWalk {
         }
     }
 
+    core::AutocorrectSuggestion suggestMethodDefinition(const core::GlobalState &gs,
+                                                        const core::Loc classOrModuleDefLoc,
+                                                        const core::MethodRef abstractMethodRef) {
+        auto [_, indentLength] = classOrModuleDefLoc.findStartOfLine(gs);
+        string indent(indentLength, ' ');
+        auto showOptions = core::ShowOptions().withUseValidSyntax().withConcretizeIfAbstract();
+
+        auto methodDefinition = core::source_generator::prettyTypeForMethod(
+            gs, abstractMethodRef, nullptr, abstractMethodRef.data(gs)->resultType, nullptr, showOptions);
+
+        vector<string> reindentedMethodDefinitionLines;
+        absl::c_transform(absl::StrSplit(methodDefinition, "\n"), std::back_inserter(reindentedMethodDefinitionLines),
+                          [indent](auto &line) -> string { return absl::StrCat(fmt::format("{}  ", indent), line); });
+
+        auto reindentedMethodDefinition = absl::StrJoin(reindentedMethodDefinitionLines, "\n");
+
+        return core::AutocorrectSuggestion{
+            fmt::format("Define `{}`", abstractMethodRef.show(gs)),
```

Comment:
I think you likely want to show the method name here, not the symbol name.

It's the difference between `Some::Really::Long::Class::Name#foo` vs `foo`.

Also, using `abstractMethodRef` will show the name of the abstract class--what you're doing here isn't defining the abstract method, but rather defining a method in the concrete class to override the abstract one, which could be confusing, unless we simply write `foo`.

Reply:
Fixed in [0dd3397](https://github.com/sorbet/sorbet/pull/7048/commits/0dd3397c7ea6b59ea065a0bc03306d1eb6b7a729).